ksql-functional-tests/src/main/java/io/confluent/ksql/test/tools/KsqlTestingTool.java

```diff
@@ -122,14 +117,16 @@ static void runWithTripleFiles(
         true
     );
 
+    final File statementFileObject = new File(statementFile);
+    final FunctionRegistry functionRegistry
+        = TestRunnerFunctionRegistryUtil.getFunctionRegistry(statementFileObject);
```

Comment:
I think this might slow down our tests (in `KsqlTestingToolTest`) because we are loading the functions every time. If the test does not supply a custom jar, I think we should fallback to `TestFunctionRegistry.INSTANCE.get()` - can you post how long it takes to run the tests before and after this change?

Reply:
(Replying to Almog's thread: https://github.com/confluentinc/ksql/pull/3252/files#r316902086)

So `TestFunctionRegistry.INSTANCE` will also be loading UDFs once per test file if the testing tool doesn't support running a directory structure of tests.

If we do as I suggest and require the UDF jar on the class path then `TestFunctionRegistry.INSTANCE` will already have loaded the custom UDFs.

We certainly shouldn't be loading UDFs twice. This is really slow!

So our fix should not be to mess around with the logic here - it should always use `TestFuncitonRegistry.INSTANCE` - and our fix for performance should be to support running the tool against recursively against a directory structure of tests.  This can either be with single _combined_ test files, or with the separate files, but matching them using some naming convention.